src/components/ui/HeadingWithQuickSearch.stories.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import { storiesOf } from '@storybook/react'
+import { action } from '@storybook/addon-actions'
+
+import State from '../../utils/State'
+
+import HeadingWithQuickSearch from './HeadingWithQuickSearch'
+
+const Heading = ({ children }) => (
+  <div style={{ height: '58px', width: '400px', border: '1px solid rgb(233, 237, 238)', backgroundColor: 'rgb(249, 250, 251)' }}>
+    { children }
+  </div>
+)
+
+storiesOf('UI Components/HeadingWithQuickSearch', module)
+  .add('self-controlled sidebar heading', () => (
+    <Heading>
+      <HeadingWithQuickSearch
+        text='Filters'
+        leftIcon='filters'
+        onQuickSearchSubmit={action('quickSearchSubmit')}
+      />
+    </Heading>
+  ))
+  .add('fully-controlled sidebar heading', () => (
+    <State initialState={{ isOpen: false, value: '' }}>
+      {({ setState, state }) => (
+        <Heading>
+          <HeadingWithQuickSearch
+            text='Filters'
+            leftIcon='filters'
+            isQuickSearching={state.isOpen}
+            quickSearchValue={state.value}
+            onToggleQuickSearch={(isOpen) => setState({ isOpen })}
+            onQuickSearchChangeValue={(value) => setState({ value })}
+          />
+        </Heading>
+      )}
+    </State>
+  ))
+
+storiesOf('UI Components/HeadingWithQuickSearch/Debug', module)
+  .add('with action handler', () => (
+    <Heading>
+      <HeadingWithQuickSearch
+        text='Click this arrow'
+        leftIcon='arrowleft'
+        leftIconAction={action('onClick')} />
+    </Heading>
+  ))
+  .add('missing props (component does not explode)', () => (
```

Comment:
Lets remove the wrapper in this particular debug case, we don't need to worry about visual context, just need to make sure that the component doesn't explode if missing props and if the "minimal render" doesn't look like a broken UI piece (many times will be fine to be blank or even null - no markup at all).

```suggestion
  .add('missing props (does component explode?)', () => (
    <HeadingWithQuickSearch />
  ))
```